config/config.go

```diff
@@ -815,6 +816,9 @@ func (c *KubernetesSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) er
 	if err := checkOverflow(c.XXX, "kubernetes_sd_config"); err != nil {
 		return err
 	}
+	if c.Kind != "nodes" && c.Kind != "service_endpoints" && c.Kind != "pods" && c.Kind != "api_servers" {
```

Comment:
Container and service are missing.

Are you changing to plural and adding _ on purpose? That'll make it trickier for users to switch.